lib/std/mem.zig

```diff
@@ -3196,7 +3196,10 @@ pub fn alignForward(addr: usize, alignment: usize) usize {
 /// Round an address up to the nearest aligned address
 /// The alignment must be a power of 2 and greater than 0.
 pub fn alignForwardGeneric(comptime T: type, addr: T, alignment: T) T {
-    return alignBackwardGeneric(T, addr + (alignment - 1), alignment);
+    var result: T = undefined;
+    // Add with overflow due to possibility of addr + alignment - 1 being larger than std.math.maxInt(T).
+    // result is clamped between 0 and std.math.maxInt(usize).
+    return alignBackwardGeneric(T, if (@addWithOverflow(T, addr, alignment - 1, &result)) std.math.maxInt(T) else result, alignment);
```

Comment:
you can use [**Saturating Subtraction**](https://ziglang.org/documentation/master/#Table-of-Operators)
```suggestion
return alignBackwardGeneric(T, addr +| (alignment - 1), alignment);
```

Reply:
and i guess here we can return `OutOfMemory`:
https://github.com/ziglang/zig/blob/9c66fdadc709ca4c00b08aa8fa4dc6312cb559c3/lib/std/os.zig#L4375

Reply:
I've now changed the `unreachable` to `OutOfMemory`. Also changed the `@subWithOverflow` with a saturating subtraction to make it more consistent.